test/e2e/e2e_test.go

```diff
@@ -292,6 +298,131 @@ func pollForConfigMapChange(client *kubernetes.Clientset, compareConfigMap *v1.C
 	})
 }
 
+// generateRoute creates a OpenShift routes
+func generateRoute(routeClient *routeclient.RouteV1Client, s *v1.Service, namespace string) (*routev1.Route, error) {
+	route, err := routeClient.Routes(namespace).Create(&routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ObjectMeta.Name,
+			Namespace: s.ObjectMeta.Namespace,
+			Labels:    s.ObjectMeta.Labels,
+		},
+		Spec: routev1.RouteSpec{
+			TLS: &routev1.TLSConfig{
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+				Termination:                   routev1.TLSTerminationPassthrough,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: s.Name,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: s.Spec.Ports[0].TargetPort,
+			},
+		},
+	})
+
+	if err = wait.PollImmediate(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		localRoute, err := routeClient.Routes(namespace).Get(s.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		if !IsRouteAdmitted(localRoute) {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+	return route, nil
+}
+
+func IsRouteAdmitted(route *routev1.Route) bool {
+	admitted := false
+	for _, ingress := range route.Status.Ingress {
+		for _, condition := range ingress.Conditions {
+			if condition.Type == routev1.RouteAdmitted && condition.Status == v1.ConditionTrue {
+				admitted = true
```

Comment:
this now loops even though it long knows `admitted = true`

Reply:
yeah, I will add a `break` statment after `admitted = true`

Reply:
well, that will then just make this loop in the wrapping for loop, you can `return true` right away